### PR TITLE
Update to use datasource.Manage and sessions.GetSessionWithAuthSettings

### PR DESCRIPTION
**Pull request by iwysiu**
I did make changes to public functions, so this theoretically counts as a breaking change, but I'd also be shocked if anyone was actually consuming this plugin as a package.

I noticed that we were using a global for the sessions so I updated that while I updated the clients, but if it's too hard to review I can separate them.

### CODE REVIEW

File: pkg/datasource/datasource.go [hunk 1]
```diff
@@ -2,119 +2,107 @@ package datasource
...
 type RequestSettings struct {
 	Region string
 }
 
-func getXrayClient(ctx context.Context, pluginContext *backend.PluginContext, requestSettings RequestSettings) (XrayClient, error) {
+func (ds Datasource) getClient(ctx context.Context, pluginContext *backend.PluginContext, requestSettings RequestSettings) (XrayClient, error) {
+	return ds.xrayClientFactory(ctx, pluginContext, requestSettings, ds.authSettings, ds.sessions)
+}
+
+func getXrayClient(ctx context.Context, pluginContext *backend.PluginContext, requestSettings RequestSettings, authSettings *awsds.AuthSettings, sessions *awsds.SessionCache) (XrayClient, error) {
```
Comment (njvrzm) [inline pkg/datasource/datasource.go:105]: nit: Here and elsewhere `backend.PluginContext`, `backend.DataSourceInstanceSettings`, `awsds.AuthSettings` and `awsds.AWSDatasourceSettings` don't need to be passed around or stored as pointers. This is a bit of a hobby horse of mine - I think using pointers where they're not necessary makes code harder to understand - so ignore me if you like 😅 but I pushed a branch [here](https://github.com/grafana/x-ray-datasource/compare/iwysiu/229..njvrzm/streamline_pointers) if you want to cherry-pick it. Entirely up to you.

**Review by njvrzm — APPROVED** [review]
This is a nice cleanup! My comments are optional.

File: pkg/datasource/datasource.go [hunk 2]
```diff
@@ -2,119 +2,107 @@ package datasource
...
+			currentRes = ds.getSingleAnalyticsQueryResult(ctx, query, &req.PluginContext)
+		case QueryGetInsights:
+			currentRes = ds.getSingleInsight(ctx, query, &req.PluginContext)
+		case QueryGetServiceMap:
+			currentRes = ds.getSingleServiceMap(ctx, query, &req.PluginContext)
+		default:
+			res.Responses[query.RefID] = backend.DataResponse{
+				Error: fmt.Errorf("unknown query type: %s", query.QueryType),
+			}
+			continue
```
[the remark this answers is not in the document]
Comment (iwysiu) [reply pkg/datasource/datasource.go]: done

File: pkg/datasource/datasource.go [hunk 3]
```diff
@@ -2,119 +2,107 @@ package datasource
...
+		}
+		res.Responses[query.RefID] = currentRes
+	}
+	return res, nil
+}
+
+func (ds Datasource) CallResource(ctx context.Context, req *backend.CallResourceRequest, sender backend.CallResourceResponseSender) error {
```
[the remark this answers is not in the document]
Comment (iwysiu) [reply pkg/datasource/datasource.go]: done

File: pkg/datasource/datasource.go (hunk 1 above)
Comment (iwysiu) [reply pkg/datasource/datasource.go]: cherry-picked!

**Pull request merged**